dashboard/app/dsl/script_dsl.rb

```diff
@@ -245,39 +234,54 @@ def level(name, properties = {})
       end
     else
       script_level = {
-        lesson: @lesson,
         levels: [level]
       }
 
+      if assessment
+        script_level[:assessment] = assessment
+      end
+      if bonus
+        script_level[:bonus] = bonus
+      end
+      if named
+        script_level[:named_level] = named
+      end
+
       if progression || challenge
         script_level[:properties] = {}
         script_level[:properties][:progression] = progression if progression
         script_level[:properties][:challenge] = true if challenge
       end
 
-      @scriptlevels << script_level
+      current_lesson_group = @lesson_groups.length - 1
+      current_lesson = @lesson_groups[current_lesson_group][:lessons].length - 1
+      @lesson_groups[current_lesson_group][:lessons][current_lesson][:script_levels] << script_level
     end
   end
 
   def variants
-    @current_scriptlevel = {levels: [], properties: {}, lesson: @lesson}
+    @current_scriptlevel = {levels: [], properties: {}}
   end
 
   def endvariants
-    @scriptlevels << @current_scriptlevel
+    current_lesson_group = @lesson_groups.length - 1
+    current_lesson = @lesson_groups[current_lesson_group][:lessons].length - 1
+    @lesson_groups[current_lesson_group][:lessons][current_lesson][:script_levels] << @current_scriptlevel
     @current_scriptlevel = nil
   end
 
   # @override
   def i18n_hash
     i18n_stage_strings = {}
-    @lessons.each do |stage|
-      i18n_stage_strings[stage[:lesson]] = {'name' => stage[:lesson]}
-    end
 
     i18n_lesson_group_strings = {}
     @lesson_groups.each do |lesson_group|
-      i18n_lesson_group_strings[lesson_group[:key]] = {'display_name' => lesson_group[:display_name]}
+      unless lesson_group[:key].nil?
```

Comment:
could avoid the double negative like this:
```suggestion
      if lesson_group[:key]
```